http/http.go

```diff
@@ -521,6 +548,39 @@ func (s Server) postLogin(w http.ResponseWriter, r *http.Request) {
 	return
 }
 
+func (s Server) postAPILogin(w http.ResponseWriter, r *http.Request) {
+	authorInput := r.FormValue("author")
+
+	encryptedAuthorByte, err := security.Encrypt([]byte(authorInput), s.Key)
+	if err != nil {
+		handleJSONError(w, r, "encrypt error", http.StatusInternalServerError, err)
+		return
+	}
+
+	cookie := http.Cookie{
+		Name:   "authorName",
+		Value:  base64.StdEncoding.EncodeToString(encryptedAuthorByte),
+		Path:   "/",
+		Secure: true,
+	}
+
+	author, _ := s.DB.ReadAuthor(authorInput)
+
+	if author.Name != "" {
+		http.SetCookie(w, &cookie)
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		err = json.NewEncoder(w).Encode("logged in")
+		if err != nil {
+			handleJSONError(w, r, "encode", http.StatusInternalServerError, err)
+		}
+		return
+	}
+
+	handleJSONError(w, r, "author doesn't exist", http.StatusUnauthorized, err)
+	return
```

Comment:
I would reverse the conditions here, having a long `if` is ugly:


```suggestion
	if author.Name == "" {
		handleJSONError(w, r, "author doesn't exist", http.StatusUnauthorized, err)
	  return
	}
	
		http.SetCookie(w, &cookie)
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(http.StatusOK)
		err = json.NewEncoder(w).Encode("logged in")
		
		if err != nil {
			handleJSONError(w, r, "encode", http.StatusInternalServerError, err)
		}


```